mllib/src/test/scala/org/apache/spark/ml/classification/DecisionTreeClassifierSuite.scala

```diff
@@ -275,6 +274,40 @@ class DecisionTreeClassifierSuite extends SparkFunSuite with MLlibTestSparkConte
     val model = dt.fit(df)
   }
 
+  test("DecisionTree should support all NumericType labels") {
+    val dfWithIntLabels = TreeTests.setMetadata(sqlContext.createDataFrame(Seq(
+      (0, Vectors.dense(0, 2, 3)),
```

Comment:
It might be less verbose to create the dataframe once and then add different types of label columns. Something like: